fontforge/python.c

```diff
@@ -4367,121 +4428,216 @@ struct flaglist strokeflags[] = {
+static char *strokekey_conv[]
+    = { "type", "contour", "angle", "cap", "join", STROKE_OPTKEYS, NULL };
+static char *strokebkey_conv[]
+    = { "type", "contour", "flags", NULL };
+
+static int Stroke_Parse(StrokeInfo *si, PyObject *args, PyObject *keywds) {
+    char *str, *type, *cap="nib", *join="nib", *rostring="layer";
+    int c, j, f, r, toknum;
     PyObject *flagtuple=NULL;
-    PyObject *poly=NULL;
-    int argcnt;
+    PyObject *nib=NULL;
 
     if ( !PySequence_Check(args) ) {
 	PyErr_Format(PyExc_TypeError, "Expected a sequence");
-return( -1 );
+	return( -1 );
     }
-    argcnt = PySequence_Size(args);
-    if ( argcnt==0 ) {
+    if ( PySequence_Size(args)==0 ) {
 	PyErr_Format(PyExc_TypeError, "Expected a name of a pen type");
-return( -1 );
+	return( -1 );
     }
+
+    InitializeStrokeInfo(si);
+    si->radius *= 2;
+    si->minorradius *= 2;
+    // Verify expected defaults
+    assert( si->rmov==srmov_layer && si->cap==lc_nib && si->join==lj_nib );
+
     PYGETSTR(PySequence_GetItem(args,0), str, -1);
-    memset(si,0,sizeof(*si));
     if ( str==NULL ) {
 	ENDPYGETSTR();
-return( -1 );
+	return( -1 );
     }
-    if ( strcmp(str,"circular")==0 ) {
-	if ( !PyArg_ParseTuple(args,"sd|ssO", &str, &width, &cap, &join, &flagtuple ) ) {
-	    ENDPYGETSTR();
-return( -1 );
+    if ( strcmp(str, "circular")==0 ) {
+	si->stroke_type = si_round;
+	if ( !PyArg_ParseTupleAndKeywords(args, keywds,
+                "sd|ssd" STROKE_OPTFORMAT, strokekey_circ, &type, &si->radius,
+		&cap, &join, &si->penangle, STROKE_OPTARGS) ) {
+	    PyErr_Clear();
+	    if ( !PyArg_ParseTupleAndKeywords(args, keywds, "sd|ssO",
+                strokebkey_circ, &type, &si->radius,
+		&cap, &join, &flagtuple) ) {
+		ENDPYGETSTR();
+		PyErr_Format(PyExc_TypeError, "Wrong parameter set for "
+		                              "'circular' nib type" );
+		return( -1 );
+	    }
 	}
-	si->stroke_type = si_std;
-	minor = width;
-	angle = 0;
-    } else if ( strcmp(str,"eliptical")==0 ) {
-	if ( !PyArg_ParseTuple(args,"sddd|ssO", &str, &width, &minor, &angle, &cap, &join, &flagtuple ) ) {
-	    ENDPYGETSTR();
-return( -1 );
+    } else if (    strcmp(str, "eliptical")==0
+                || strcmp(str, "elliptical")==0 ) {
+	si->stroke_type = si_round;
+	if ( !PyArg_ParseTupleAndKeywords(args, keywds,
+                "sdd|dss" STROKE_OPTFORMAT, strokekey_ellip, &type,
+	        &si->radius, &si->minorradius, &si->penangle, &cap, &join,
+	        STROKE_OPTARGS) ) {
+	    PyErr_Clear();
+	    if ( !PyArg_ParseTupleAndKeywords(args, keywds, "sddd|ssO",
+                strokebkey_ellip, &type, &si->radius, &si->minorradius,
+		&si->penangle, &cap, &join, &flagtuple) ) {
+		ENDPYGETSTR();
+		PyErr_Format(PyExc_TypeError, "Wrong parameter set for "
+		                              "'elliptical' nib type" );
+		return( -1 );
+	    }
 	}
-	si->stroke_type = si_std;
-    } else if ( strcmp(str,"caligraphic")==0 || strcmp(str,"square")==0 ) {
```

Comment:
Is `square` no longer supported?

Reply:
I don't think `square` was ever documented but since it didn't get any special handling before it's easy enough to throw another `strcmp()` to the new list.

Reply:
The same goes for `poly`.